src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -91,6 +91,17 @@ public String linkFriends(Long accountId, Pageable pageable) {
         return linkAllAccountsWithBots(pageable);
     }
 
+    @Override
+    public FriendDto getFriendInfo(Long accountId) {
+        Account loggedAccount = getLoggedAccount();
+        Account friendAccount = getAccountById(accountId);
+        if (hasFriends(loggedAccount) && friendExist(loggedAccount, friendAccount)) {
```

Comment:
Isn't enough just to use second check `friendExist()`?

Reply:
No, because if the list of friends is null, the .contains will throw NPE

Reply:
You should fix that in friendExist()

Reply:
Now, after the fix in `friendExist()`, no need of `hasFriends(loggedAccount)` check here